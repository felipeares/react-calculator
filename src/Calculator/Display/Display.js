import React from "react";
import styles from "./Display.module.css";
import PropTypes from "prop-types";

const Display = props => (
  <div className={styles.Display}>{props.displayValue}</div>
);

Display.propTypes = { displayValue: PropTypes.string.isRequired };

export default Display;
