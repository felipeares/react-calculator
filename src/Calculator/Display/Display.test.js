import React from "react";
import { shallow } from "enzyme";
import Display from "./Display";
import PropTypes from "prop-types";

describe("Display", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Display displayValue={""} />);
  });

  it("have not changed the UI", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a main div", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should have a Display", () => {
    expect(wrapper.find("div").hasClass("Display")).toEqual(true);
  });

  it("renders the value of displayValue", () => {
    wrapper.setProps({ displayValue: "test" });
    expect(wrapper.text()).toEqual("test");
  });

  it("should check for displayValue prop to be string", () => {
    const prop_type_is_string = Display.propTypes.displayValue;
    expect(prop_type_is_string).toBe(PropTypes.string.isRequired);
  });
});
