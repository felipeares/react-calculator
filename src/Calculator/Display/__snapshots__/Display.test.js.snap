// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display have not changed the UI 1`] = `
<div
  className="Display"
/>
`;
